read-fonts/src/tables/aat.rs

```diff
@@ -0,0 +1,619 @@
+                let mut cursor = FontData::new(lookup.values_data()).cursor();
+                cursor.advance_by(offset);
+                let val = match unit_size {
+                    1 => cursor.read::<u8>()? as u32,
+                    2 => cursor.read::<u16>()? as u32,
+                    4 => cursor.read::<u32>()?,
+                    _ => {
+                        return Err(ReadError::MalformedData(
+                            "invalid unit_size in format 10 AAT lookup table",
+                        ))
+                    }
+                };
+                Ok(T::from_u32(val))
+            }
+        }
+    }
+}
+
+impl<'a, T> FontRead<'a> for TypedLookup<'a, T> {
+    fn read(data: FontData<'a>) -> Result<Self, ReadError> {
+        Ok(Self {
+            lookup: Lookup::read(data)?,
+            _marker: std::marker::PhantomData,
+        })
+    }
+}
+
+#[cfg(feature = "traversal")]
+impl<'a, T> SomeTable<'a> for TypedLookup<'a, T> {
+    fn type_name(&self) -> &str {
+        "TypedLookup"
+    }
+
+    fn get_field(&self, idx: usize) -> Option<Field<'a>> {
+        self.lookup.get_field(idx)
+    }
+}
+
+/// Trait for values that can be read from lookup tables.
+pub trait LookupValue: Copy + Scalar + bytemuck::AnyBitPattern {
+    fn from_u16(v: u16) -> Self;
+    fn from_u32(v: u32) -> Self;
+}
+
+impl LookupValue for u16 {
+    fn from_u16(v: u16) -> Self {
+        v
+    }
+
+    fn from_u32(v: u32) -> Self {
+        // intentionally truncates
+        v as _
+    }
+}
+
+impl LookupValue for u32 {
+    fn from_u16(v: u16) -> Self {
+        v as _
+    }
+
+    fn from_u32(v: u32) -> Self {
+        v
+    }
+}
+
+impl LookupValue for GlyphId {
+    fn from_u16(v: u16) -> Self {
+        GlyphId::from(v)
+    }
+
+    fn from_u32(v: u32) -> Self {
+        // intentionally truncates
+        GlyphId::from(v as u16)
+    }
+}
+
+pub type LookupU16<'a> = TypedLookup<'a, u16>;
+pub type LookupU32<'a> = TypedLookup<'a, u32>;
+pub type LookupGlyphId<'a> = TypedLookup<'a, GlyphId>;
+
+/// Entry in an (extended) state table.
+pub struct StateEntry<T = ()> {
+    /// Index of the next state.
+    pub new_state: u16,
+    /// Flag values are table specific.
+    pub flags: u16,
+    /// Payload is table specific.
+    pub payload: T,
+}
+
+impl<'a, T: bytemuck::AnyBitPattern + FixedSize> FontRead<'a> for StateEntry<T> {
+    fn read(data: FontData<'a>) -> Result<Self, ReadError> {
+        let mut cursor = data.cursor();
+        let new_state = cursor.read()?;
+        let flags = cursor.read()?;
+        let remaining = cursor.remaining().ok_or(ReadError::OutOfBounds)?;
+        let payload = *remaining.read_ref_at(0)?;
+        Ok(Self {
+            new_state,
+            flags,
+            payload,
+        })
+    }
+}
+
+impl<T> FixedSize for StateEntry<T>
+where
+    T: FixedSize,
+{
+    // Two u16 fields + payload
+    const RAW_BYTE_LEN: usize = 4 + T::RAW_BYTE_LEN;
```

Comment:
just for maximum explicitness I'd write,

```suggestion
    const RAW_BYTE_LEN: usize = u16::RAW_BYTE_LEN + u16::RAW_BYTE_LEN + T::RAW_BYTE_LEN;
```